rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java

```diff
@@ -510,6 +512,40 @@ public User convertToEmbeddedUser(User user) {
         return embeddedUser;
     }
 
+    public void addEmbeddedObligations(HalResource<License> halLicense, List<Obligation> obligations) {
+        for (Obligation obligation : obligations) {
+            HalResource<Obligation> obligationHalResource = addEmbeddedObligation(obligation);
+            halLicense.addEmbeddedResource("sw360:obligations", obligationHalResource);
+        }
+    }
+
+    public HalResource<Obligation> addEmbeddedObligation(Obligation obligation) {
+        Obligation embeddedObligation = convertToEmbeddedObligation(obligation);
+        HalResource<Obligation> halObligation = new HalResource<>(embeddedObligation);
+        try {
+            Link obligationSelfLink = linkTo(UserController.class)
+                    .slash("api" + ObligationController.OBLIGATION_URL + "/" + obligation.getId()).withSelfRel();
+            halObligation.add(obligationSelfLink);
+            return halObligation;
+        } catch (Exception e) {
+            LOGGER.error("cannot create self link for obligation with id: " +obligation.getId());
+        }
+        return null;
+    }
+
+    public Obligation convertToEmbeddedObligation(Obligation obligation) {
```

Comment:
May be `convertToEmbeddedObligation(Obligation obligation)` & `convertToEmbeddedObligation(String title)` can be merged to a single method.

Reply:
Function `convertToEmbeddedObligation(String title)` was removed.